nesta_ds_utils/file_ops.py

```diff
@@ -1,53 +1,431 @@
+    """
+    if fnmatch(path_from, "*.csv"):
+        np_array_data = np.genfromtxt(fileobj, delimiter=",", **kwargs)
+    elif fnmatch(path_from, "*.parquet"):
+        np_array_data = pq.read_table(fileobj, **kwargs)["data"].to_numpy()
+
+    return np_array_data
+
+
+def download_obj(
+    bucket: str,
+    path_from: str,
+    download_as: str = None,
+    kwargs_boto: dict = {},
+    kwargs_reading: dict = {},
+) -> any:
+    """Download data to memory from S3 location.
+
+    Args:
+        bucket (str): Bucket's name.
+        path_from (str): Path to data in S3.
+        download_as (str, optional): Type of object downloading. Choose between
+        ('dataframe', 'dict', 'list', 'str', 'np.array'). Not needed for 'pkl files'.
+        kwargs_boto (dict, optional): Dictionary of kwargs for boto3 function 'download_fileobj'.
+        kwargs_reading (dict, optional): Dictionary of kwargs for reading data.
+
+    Returns:
+        any: Donwloaded data.
+    """
+    if not path_from.endswith(tuple([".csv", ".parquet", ".json", ".txt", ".pkl"])):
+        raise Exception(
+            "This file type is not currently supported for download in memory."
+        )
+    s3 = boto3.client("s3")
+    fileobj = io.BytesIO()
+    s3.download_fileobj(bucket, path_from, fileobj, **kwargs_boto)
+    fileobj.seek(0)
+    if download_as == "dataframe":
+        if path_from.endswith(tuple([".csv", ".parquet"])):
+            return _fileobj_to_df(fileobj, path_from, **kwargs_reading)
+        else:
+            raise Exception(
+                "Download as dataframe currently supported only "
+                "for 'csv' and 'parquet'."
+            )
+    elif download_as == "dict":
+        if path_from.endswith(tuple([".json"])):
+            return _fileobj_to_dict(fileobj, path_from, **kwargs_reading)
+        else:
+            raise Exception(
+                "Download as dictionary currently supported only " "for 'json'."
+            )
+    elif download_as == "list":
+        if path_from.endswith(tuple([".csv", ".txt", ".json"])):
+            return _fileobj_to_list(fileobj, path_from, **kwargs_reading)
+        else:
+            raise Exception(
+                "Download as list currently supported only "
+                "for 'csv', 'txt' and 'json'."
+            )
+    elif download_as == "str":
+        if path_from.endswith(tuple([".txt"])):
+            return _fileobj_to_str(fileobj)
+        else:
+            raise Exception("Download as string currently supported only " "for 'txt'.")
+    elif download_as == "np.array":
+        if path_from.endswith(tuple([".csv", ".parquet"])):
+            return _fileobj_to_np_array(fileobj, path_from, **kwargs_reading)
+        else:
+            raise Exception(
+                "Download as numpy array currently supported only "
+                "for 'csv' and 'parquet'."
+            )
+    elif not download_as:
+        if path_from.endswith(tuple([".pkl"])):
```

Comment:
```suggestion
        if path_from.endswith(".pkl"):
```
Can just be the string if one option.